superset-frontend/src/SqlLab/components/SqlEditor/index.jsx

```diff
@@ -673,6 +684,15 @@ const SqlEditor = ({
         onDragEnd={onResizeEnd}
       >
         <div ref={northPaneRef} className="north-pane">
+          {SqlFormExtension && (
+            <SqlFormExtension
+              queryEditorId={queryEditor.id}
+              setQueryEditorAndSaveSqlWithDebounce={
```

Comment:
@geido @Antonio-RiveroMartnez similar to [the previous comment](https://github.com/apache/superset/pull/24205#discussion_r1205807291), this action props doesn't make sense since your custom component only needs these actions. You can hook up these actions in your custom component by combining of existing action helpers

For example you can build startQuery in your component by combining of runQueryFromSqlEditor and setActiveSouthPaneTab
```
      dispatch(
        runQueryFromSqlEditor(
          database,
          queryEditor,
          defaultQueryLimit,
          ctasArg ? ctas : '',
          ctasArg,
          ctas_method,
        ),
      );
      dispatch(setActiveSouthPaneTab('Results'));
```

Reply:
Hey @justinpark sure, I can recreate all those with the helpers, but since we are not changing how they work, just invoking them, I treated them like other props our existing extensions are using. IMHO I see no major benefit by forcing a duplication of code in the extension. However, if you truly see this like an scenario we should force such duplication, I can work on it in a follow up PR. Thanks as always for the input.